pandas/core/algorithms.py

```diff
@@ -395,6 +395,14 @@ def isin(comps, values):
                         " to isin(), you passed a [{values_type}]"
                         .format(values_type=type(values).__name__))
 
+    # GH 25507
+    # if `values` is a set, directly use it instead of hashing a list
+    if is_set_like(values):
+        result = np.empty_like(comps, dtype=np.bool)
+        for i, comp in enumerate(comps):
+            result[i] = comp in values
```

Comment:
for each test in test_algorithms for isin, add an arg that also is a set (for as many tests as possible)